### PR TITLE
homework11

### PR DESCRIPTION
**Pull request by VeronicaZhigalova**

### CODE REVIEW

File: src/main/java/org/example/Homework11.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package org.example;
+
+import java.util.*;
+
+public class Homework11 {
+
+    public Integer maxDuplicate(List<Integer> nums) {
+        List<Integer> uniqueNumbers = new ArrayList<>();
```
Comment (tallinn-learning-ilja) [inline src/main/java/org/example/Homework11.java]: Better use Set for uniqueNumbers

File: src/main/java/org/example/Homework11.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        }
+        if (duplicates.isEmpty()) {
+            return null;
+        }
+        return (Integer) Collections.max(duplicates);
+    }
+
+    public List<String> wordNumeration(List<String> words) {
+        List<String> numberedWords = new ArrayList<>();
+        HashMap wordCounts = new HashMap<>();
```
Comment (tallinn-learning-ilja) [inline src/main/java/org/example/Homework11.java]: Not `HashMap` but `Map<String, Integer>`. Raw usage is discouraged, look at the Intellij's warnings as well. Also, it's a better practice to use Interface as a type

File: src/test/java/org/example/Homework11Test.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+public class Homework11Test {
+
+    private final Homework11 homework11 = new Homework11();
+
+    @Test
+    void testMaxDuplicates() {
+        List<Integer> numbers = List.of(1, 2, 2, 1, 1);
+        int actualResult = homework11.maxDuplicate(numbers);
+        assertEquals(2, actualResult);
+    }
```
Comment (tallinn-learning-ilja) [inline src/test/java/org/example/Homework11Test.java:19]: The point of tests is to try out different scenarios. Add tests for the case, when the list is empty, when there are no duplicates

File: src/test/java/org/example/Homework11Test.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        List<String> actualResult = homework11.wordNumeration(words);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void testMakeBricks() {
+        boolean bricksResult = homework11.makeBricks(3, 2, 10);
+        assertEquals(true, bricksResult);
+
+    }
```
Comment (tallinn-learning-ilja) [inline src/test/java/org/example/Homework11Test.java]: Here as well, more cases when it's not possible to make a wall

**Review by tallinn-learning-ilja — CHANGES_REQUESTED** [review]
Couple things + add more tests

File: src/main/java/org/example/Homework11.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        return (Integer) Collections.max(duplicates);
+    }
+
+    public List<String> wordNumeration(List<String> words) {
+        List<String> numberedWords = new ArrayList<>();
+        Map<String, Integer> wordCounts = new HashMap<>();
+        for (String word : words) {
+            String wordWord = word.toLowerCase();
+            int count = (int) wordCounts.getOrDefault(wordWord, 0) + 1;
```
Comment (tallinn-learning-ilja) [inline src/main/java/org/example/Homework11.java:31]: You don't need to cast it to int, it's done automatically

**Review by tallinn-learning-ilja — APPROVED** [review]
Small thing but overall good work
